include/ccf/rpc_context.h

```diff
@@ -116,7 +116,10 @@ namespace enclave
     {
       nlohmann::json body = ccf::ODataErrorResponse{
         ccf::ODataError{std::move(error.code), std::move(error.msg)}};
-      const auto s = body.dump();
+      // Set error_handler to replace, to avoid throwing if the error message
+      // contains non-UTF8 characters. Other args are default values
+      const auto s =
+        body.dump(-1, ' ', false, nlohmann::json::error_handler_t::replace);
```

Comment:
We ought to have a test in the logging sample where we return a non UTF-8 message, to check that this does in fact do what we think it does.

Reply:
This was tested in some of the e2e_logging variations - anything that returns a 404 Not Found rather than a 400 Bad Request includes a re-encoding of a non-UTF8 message (and before this change, converting to a 500 as this line exploded).

But I've made that more clear, with an explicit new test in `e2e_logging`:
```
    # Tests non-UTF8 encoding in OData
    send_bad_raw_content(b"POST /node/\xff HTTP/2.0\r\n\r\n")
```